kernel/core/include/os.ads

```diff
@@ -63,8 +63,7 @@ is
    subtype os_status_t is types.int32_t
                            range OS_ERROR_MAX .. OS_SUCCESS;
 
-   subtype os_mbx_index_t is types.uint8_t
-                           range 0 .. OS_MAX_MBX_ID;
+   type os_mbx_index_t is mod OS_MAX_MBX_ID;
```

Comment:
Does moving to a modular type keeps the 8 bits (char) size of the type?

Reply:
Not necessary, it's up to the compiler.
To force it, add:
for os_mbx_index_t'size use 8;
or with aspect flavor:
type os_mbx_index_t is mod OS_MAX_MBX_ID
   with Size => 8;

Reply:
I resolved the problem by splitting the os_task_rw element in 2 parts.
- one part dedicated to the C part and unknown to the Ada part
- one part dedicated to the Ada part and unknown from the C part.

This way we can let Ada optimize the data size as it see fit without requiring data compatibility with the C part.